analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AvoidUnderPosting.cs

```diff
@@ -0,0 +1,141 @@
+        KnownType.Microsoft_AspNetCore_Http_IFormFile,
+        KnownType.Microsoft_AspNetCore_Http_IFormFileCollection);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(compilationStart =>
+        {
+            if (!compilationStart.Compilation.ReferencesControllers())
+            {
+                return;
+            }
+            var examinedTypes = new ConcurrentDictionary<ITypeSymbol, bool>();
+
+            compilationStart.RegisterSymbolStartAction(symbolStart =>
+            {
+                var type = (INamedTypeSymbol)symbolStart.Symbol;
+                if (type.IsControllerType())
+                {
+                    symbolStart.RegisterSyntaxNodeAction(nodeContext => ProcessControllerMethods(nodeContext, examinedTypes), SyntaxKind.MethodDeclaration);
+                }
+            }, SymbolKind.NamedType);
+        });
+
+    private static void ProcessControllerMethods(SonarSyntaxNodeReportingContext context, ConcurrentDictionary<ITypeSymbol, bool> examinedTypes)
+    {
+        if (context.SemanticModel.GetDeclaredSymbol(context.Node) is IMethodSymbol method
+            && method.IsControllerActionMethod())
+        {
+            var modelParameterTypes = method.Parameters
+                .Where(x => !HasValidateNeverAttribute(x))
+                .Select(x => x.Type)
+                .OfType<INamedTypeSymbol>()
+                .SelectMany(RelatedTypesToExamine)
+                .Distinct();
+            foreach (var modelParameterType in modelParameterTypes)
+            {
+                CheckInvalidProperties(modelParameterType, context, examinedTypes);
+            }
+        }
+    }
+
+    private static void CheckInvalidProperties(INamedTypeSymbol parameterType, SonarSyntaxNodeReportingContext context, ConcurrentDictionary<ITypeSymbol, bool> examinedTypes)
+    {
+        var declaredProperties = new List<IPropertySymbol>();
+        GetAllDeclaredProperties(parameterType, examinedTypes, declaredProperties);
+        var invalidProperties = declaredProperties.Where(x => !CanBeNull(x.Type)
+            && !x.HasAttribute(KnownType.System_ComponentModel_DataAnnotations_RequiredAttribute));
+        foreach (var property in invalidProperties)
+        {
+            var propertySyntax = property.DeclaringSyntaxReferences[0].GetSyntax();
+            context.ReportIssue(Rule, propertySyntax.GetLocation());
+        }
+    }
+
+    private static bool IgnoreType(ITypeSymbol type) =>
+        type is not INamedTypeSymbol namedType
+        || namedType.IsAny(IgnoredTypes)
+        || namedType.IsTupleType()
+        || (!namedType.IsRecord()
+            && !namedType.IsValueType
+            && !namedType.IsInterface()
+            && !namedType.Is(KnownType.System_String)
```

Comment:
`System.String` doesn't have a default constructor, but it has a customer binder.